File: Rosu/programmazione/laboratorio/java/problemi/problema12/src/ro/razvan/uniud/nQueensWithGui/java/Principale.java
package ro.razvan.uniud.nQueensWithGui.java;

import org.jetbrains.annotations.NotNull;
import queens.ChessboardView;

public final class Principale {

    public static void main(@NotNull final String[] args) {

        final var size = Integer.parseInt(args[0]);

        final var gui = new ChessboardView(size);
        final var solutions = Queens.listOfAllSolutions(size);

        Queens.displaySolutions(gui, solutions);

    }

}

File: Rosu/programmazione/laboratorio/java/problemi/problema12/src/ro/razvan/uniud/nQueensWithGui/kotlin/Principale.kt
package ro.razvan.uniud.nQueensWithGui.kotlin

import queens.ChessboardView
import ro.razvan.uniud.nQueensWithGui.kotlin.Queens.displaySolutions

fun main(args: Array<String>) {

    val size = args[0].toInt()

    val gui = ChessboardView(size)
    val solutions = Queens.listOfAllSolutions(size)

    gui.displaySolutions(solutions)

}